operations_on_datastructures/circular_linked_list.cpp

```diff
@@ -1,97 +1,275 @@
-#include <iostream>
-using namespace std;
+/**
+ * @file
+ * @brief Implementation for a [Circular Linked
+ * List](https://www.geeksforgeeks.org/circular-linked-list/).
+ * @details A Circular Linked List is a variation on the regular linked list, in
+ * which the last node has a pointer to the first node, which creates a full
+ * circle. Consequently, this allows any node to be used as the starting point
+ * for the list.
+ * @author [Alvin](https://github.com/polarvoid)
+ */
 
-struct node {
-    int val;
-    node *next;
-};
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+#include <vector>    /// for std::vector
 
-node *start;
+/**
+ * @namespace operations_on_datastructures
+ * @brief Operations on Data Structures
+ */
+namespace operations_on_datastructures {
 
-void insert(int x) {
-    node *t = start;
+/**
+ * @namespace circular_linked_list
+ * @brief Functions for the Circular Linked List implementation
+ */
+namespace circular_linked_list {
 
-    if (start != NULL) {
-        while (t->next != start) {
-            t = t->next;
-        }
-        node *n = new node;
-        t->next = n;
-        n->val = x;
-        n->next = start;
-    } else {
-        node *n = new node;
-        n->val = x;
-        start = n;
-        n->next = start;
+/**
+ * @brief A Node struct that represents a single Node in a Binary Tree
+ */
+struct Node {
+    int64_t data;  ///< The value of the Node
+    Node* next;    ///< The Node's successor
+    /**
+     * @brief Creates a new Node with some initial data
```

Comment:
Missing parameter documentation (`_data`).

Reply:
Oooh, I missed that, thanks :)